src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[fromFormat.length - 1];
+  const dateParts = date.split(separator);
+  const dateComponents = {};
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  fromFormat.forEach((el, index) => {
+    if (el !== separator) {
+      dateComponents[el] = dateParts[index];
+    }
+  });
+
+  if (dateComponents['YY']) {
+    dateComponents['YYYY'] =
+      dateComponents['YY'] < 30
+        ? '20' + dateComponents['YY']
+        : '19' + dateComponents['YY'];
+  }
+
+  if (dateComponents['YYYY'] && toFormat.includes('YY')) {
+    dateComponents['YY'] = dateComponents['YYYY'].slice(-2);
+  }
+
+  const formattedDate = toFormat
+    .filter((el) => el !== newSeparator)
+    .map((el) => dateComponents[el])
+    .join(newSeparator);
+
+  return formattedDate;
```

Comment:
Consider adding input validation to ensure that the `date`, `fromFormat`, and `toFormat` arguments are provided in the expected format and that they contain valid values. This can prevent potential runtime errors and make the function more robust.